scripts/generate-gql-error-index-from-template.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python3
+
+import os
+import re
+from pathlib import Path
+import argparse
+
+# For debugging - examine a single file to understand its structure
+def examine_error_file(file_path):
+    """Print the contents of an error file for debugging."""
+    try:
+        with open(file_path, 'r') as f:
+            content = f.read()
+            print(f"=== File Content for {file_path.name} ===")
+            print(content)
+            print("===================================")
+
+            # Try to extract the description with different patterns
+            patterns = [
+                r'Status description::\s*(.*?)(?=\n\n|\n==|\Z)',
+                r'== Status description\s*\n(.*?)(?=\n\n|\n==|\Z)',
+                r'== Status description\s*\n\s*Status description:: (.*?)(?=\n\n|\n==|\Z)'
+            ]
+
+            for i, pattern in enumerate(patterns):
+                match = re.search(pattern, content, re.DOTALL)
+                if match:
+                    print(f"Pattern {i+1} matched! Description: {match.group(1)[:50]}...")
+                else:
+                    print(f"Pattern {i+1} did not match")
+    except Exception as e:
+        print(f"Error reading file: {e}")
+
+def get_error_codes_from_files(errors_dir, verbose=False):
+    """Get all error codes and their details from individual .adoc files."""
+    error_codes = {}
+    sample_file = None
+
+    for file in os.listdir(errors_dir):
+        if file.endswith('.adoc') and file != 'index.adoc' and file != 'auto-index.adoc':
+            error_code = file[:-5]  # Remove .adoc extension
+            file_path = os.path.join(errors_dir, file)
+
+            if sample_file is None:
+                sample_file = file_path  # Save first file for debugging
+
+            # Extract details from file
+            with open(file_path, 'r') as f:
+                content = f.read()
+
+                # Extract page role
+                page_role = None
+                role_match = re.search(r':page-role:\s*(.*?)(?=\n|\Z)', content)
+                if role_match:
+                    page_role = role_match.group(1).strip()
+
+                # Extract description - try several patterns
+                description = None
+                desc_patterns = [
```

Comment:
If I understand this correctly it parses the individual files for each GqlStatus
- `22NA5.adoc`
- `42I33.adoc`
- etc

and produces the `index.adoc` page?

This regex magic seems fragile. An alternative solution could be to store a table `gqlcodes.csv` and produce both the `index.adoc` as well as all the individual pages from it. Ideally the Neo4j build would produce the `gqlcodes.csv` for this.

Reply:
Hmm, we can talk about this. Off the top of my head, I would prefer to simplify the regex than use the codebase to produce the individual files. Also, in those individual files, we'll have explanations, solutions, examples, etc.

Reply:
And, most importantly, doc's formatting.